specutils/analysis/width.py

```diff
@@ -212,11 +228,36 @@ def _compute_gaussian_sigma_width(spectrum, regions=None):
 
     if flux.ndim > 1:
         spectral_axis = np.broadcast_to(spectral_axis, flux.shape, subok=True)
+        centroid_result_uncert = centroid_result.uncertainty
         centroid_result = centroid_result[:, np.newaxis]
+        centroid_result.uncertainty = centroid_result_uncert[:, np.newaxis] if centroid_result_uncert is not None else None  # noqa
```

Comment:
This is where the centroid uncertainties were being dropped for multiple spectra.  Note that cases like this are just examples of why attaching uncertainties directly to the quantities is not an ideal solution as they are easily dropped (either internally or by users) and the workaround is a bit... ugly.